### PR TITLE
Ft avail requests for approval 167735050

### PR DESCRIPTION
**Pull request by cleave3**
**What does this PR do?**

  If merged this PR will create endpoint for manager to view requests of his/her direct reports

  **Description of Task to be completed?**

  Manager should be able to view open travel request of only his/her direct report

  **How should this be manually tested?**

  1. Run npm run db:migration
  2. Run npm run seed
  3. Login as manager
  4. On postman, add  token to headers
  5. On postman send a GET request to api/v1/requests/pending

  **Any background context you want to provide?**

seeded managers logins 
email: cleave@mail.com, password: cleave12345, 
email: cleave4@mail.com, password: cleave12345

  **What are the relevant stories?**

  #33 167735050

  **Screenshots (if appropriate)**

  
image

  **Questions:**

 xxx

### CODE REVIEW

File: Procfile [hunk 1]
```diff
@@ -0,0 +1,2 @@
+release: node_modules/.bin/sequelize db:migrate
+web: npm start
```
Comment (SamuelKoroh) [inline Procfile]: Add a line break here

Comment (cleave3) [reply Procfile]: ok thanks. i will do that

File: src/tests/request.test.js [hunk 2]
```diff
@@ -262,6 +293,40 @@ describe('/api/v1/requests', () => {
       res.should.have.status(404);
       expect(JSON.parse(res.text).error).to.equal('You\'ve not made any requests');
     });
+
+    it('should retrieve all open requests made by manager\'s direct report', async () => {
+      const res = await chai.request(app)
+        .get(`${URL_PREFIX}/pending`)
+        .set('token', manager.body.data.token);
+
+      res.should.have.status(200);
```
Comment (Annmary12) [inline src/tests/request.test.js:302]: Check `res.body.data` length of array to ensure that it is returning the actually amount of data

Comment (cleave3) [reply src/tests/request.test.js:302]: OK, I will do that

File: src/services/requestService.js [hunk 3]
```diff
@@ -82,9 +87,44 @@ export default class RequestService {
   }
 
   /**
-     * @param {object} details - user trip details
-     * @returns {object} trip - details
-     */
+   *
+   * @param {number} id - manager's id
+   * @return {obeject} - open requests
```
Comment (elikeyz) [inline src/services/requestService.js]: Please correct the spelling from `obeject` to `object`

File: src/docs/docs.yaml [hunk 4]
```diff
@@ -476,9 +475,9 @@ paths:
                 type: string
                 description: Trip type
                 enum:
-                  - one-way
+                  - oneway
```
Comment (elikeyz) [inline src/docs/docs.yaml]: Please revert this change, it goes against what already exists in the Request Model

File: src/docs/docs.yaml [hunk 5]
```diff
@@ -476,9 +475,9 @@ paths:
...
                   - return
-                  - multi-city
+                  - multicity
```
Comment (elikeyz) [inline src/docs/docs.yaml]: Please revert this also, it goes against the enum in the Request Model

File: src/docs/docs.yaml [hunk 6]
```diff
@@ -308,11 +308,6 @@ paths:
...
       produces:
         - application/json
-      parameters:
-        - in: header
-          name: token
-          description: Authenticate User
-          required: true
       responses:
         404:
           description: You've not make any requests
```
Comment (elikeyz) [inline src/docs/docs.yaml]: Please correct this to `You've not made any requests`

Comment (cleave3) [reply src/docs/docs.yaml]: thanks, i will do that

File: src/docs/docs.yaml (hunk 5 above)
Comment (cleave3) [reply src/docs/docs.yaml]: thanks, i will do that

File: src/docs/docs.yaml (hunk 4 above)
Comment (cleave3) [reply src/docs/docs.yaml]: thanks, i will do that

File: src/services/requestService.js (hunk 3 above)
Comment (cleave3) [reply src/services/requestService.js]: thanks, i will do that

File: src/routes/request.js [hunk 7]
```diff
@@ -11,13 +12,16 @@ const {
...
   multiCityRequest,
   getRequests,
+  availOpenRequests,
   returnRequest
 } = RequestController;
 
 router.post('/multi-city', [auth, validator(requestSchema)], multiCityRequest);
 router.post('/one-way', [auth, validator(requestSchema)], oneWay);
 router.patch('/:requestId', [auth, validator(requestSchema)], update);
+router.get('/pending', auth, permitUser(['manager']), availOpenRequests);
```
Comment (amoskeyz) [inline src/routes/request.js:22]: a `super admin` should be able to access this route too. `permitUser['manager', 'super_admin']`

Comment (cleave3) [reply src/routes/request.js:22]: @amoskeyz this is a manager's role. I don't think this suits a super admin unless he would be assigned as manager to a department as well. I think a super admin should have another route where he can get all requests not just for direct reports. Do you see my point ,?

Comment (amoskeyz) [reply src/routes/request.js:22]: @cleave3 yea i do, but according to the PO she said a super admin should be able to access all route in the app.

Comment (amoskeyz) [reply src/routes/request.js:22]: ok... i get the point now
